tigris/collection_test.go

```diff
@@ -162,6 +229,42 @@ func TestCollectionBasic(t *testing.T) {
 	it, err = c.ReadAll(ctx, fields.All)
 	require.NoError(t, err)
 
+	rit := mock.NewMockSearchResultIterator(ctrl)
+
+	// search with all params parses completely
+	sr := search.NewRequestBuilder("search query").WithSearchFields("field_1").
+		WithFilter(filter.Eq("field_2", "some value")).
+		WithFacet(search.NewFacetQueryBuilder().WithFields("field_3").Build()).
+		WithReadOptions(*fields.Include("field_4")).
+		WithOptions(&search.DefaultRequestOptions).
+		Build()
+	mdb.EXPECT().Search(ctx, "coll_1", &driver.SearchRequest{
+		Q:            sr.Q,
+		SearchFields: sr.SearchFields,
+		Filter:       driver.Filter(`{"field_2":{"$eq":"some value"}}`),
+		Facet:        driver.Facet(`{"field_3":{"size":10}}`),
+		ReadFields:   driver.Projection(`{"field_4":true}`),
+		Page:         sr.Options.Page,
+		PageSize:     sr.Options.PerPage,
+	}).Return(rit, nil)
+	searchIter, err := c.Search(ctx, sr)
+	require.NoError(t, err)
+	require.NotNil(t, searchIter)
+
+	// TODO: Add tests for validating iterator conversion
+	//type collType Coll1
+	//var r driver.SearchResponse
+	//rit.EXPECT().Next(&r).SetArg(0, createSearchResponse(t, d1)).Return(true)
+	//rit.EXPECT().Next(&r).Return(false)
+	//rit.EXPECT().Err().Return(nil)
+
+	//var rs search.Result[collType]
+	//for searchIter.Next(&rs) {
```

Comment:
This test is incomplete because I was getting a `Cannot use '&rs' (type *search.Result[collType]) as the type *search.Result[T]` warning. Maybe I misunderstood generics and did not set it properly.

```go
type SearchIterator[T interface{}] struct {}
func (it *SearchIterator[T]) Next(res *search.Result[T]) bool {}
```